code/modules/reagents/machinery/dispenser/reagent_tank.dm

```diff
@@ -10,14 +10,32 @@
 	anchored = 0
 	pressure_resistance = 2*ONE_ATMOSPHERE
 
+	var/modded = FALSE
+
 	var/obj/item/hose_connector/input/active/InputSocket
 	var/obj/item/hose_connector/output/active/OutputSocket
 
 	var/amount_per_transfer_from_this = 10
 	var/possible_transfer_amounts = list(10,25,50,100)
 
 /obj/structure/reagent_dispensers/attackby(obj/item/weapon/W as obj, mob/user as mob)
-		return
+	src.add_fingerprint(user)
+	if (W.is_wrench())
+		user.visible_message("[user] wrenches [src]'s faucet [modded ? "closed" : "open"].", \
+			"You wrench [src]'s faucet [modded ? "closed" : "open"]")
+		modded = modded ? 0 : 1
+		playsound(src, W.usesound, 75, 1)
+		if (modded)
+			message_admins("[key_name_admin(user)] opened a reagent tank at [loc.loc.name] ([loc.x],[loc.y],[loc.z]), leaking [LAZYLEN(reagents.reagent_list) ? english_list(reagents.reagent_list) : "nothing"]. (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[loc.x];Y=[loc.y];Z=[loc.z]'>JMP</a>)")
+			log_game("[key_name(user)] opened reagent tank at [loc.loc.name] ([loc.x],[loc.y],[loc.z]), leaking [LAZYLEN(reagents.reagent_list) ? english_list(reagents.reagent_list) : "nothing"].")
+			leak(amount_per_transfer_from_this)
+	return ..()
+
+/obj/structure/reagent_dispensers/examine(mob/user)
+	. = ..()
+	if(get_dist(user, src) <= 2)
+		if(modded)
+			. += SPAN_WARNING("The faucet is wrenched open, leaking the contents!")
```

Comment:
If this is what `modded` is actually tracking, then please update the var name to reflect this. It's... really generic, and not immediately obvious.